src/store/getters/state-getters.ts

```diff
@@ -517,62 +517,84 @@ export const hasContactInfoChanged = (state: StateIF): boolean => {
   )
 }
 
-/** The dynamic Office Type based on the current filing. */
-export const officeType = (state: StateIF): OfficeTypes => {
-  if (isChangeFiling(state) || isConversionFiling(state)) return OfficeTypes.BUSINESS_OFFICE
-  if (isAlterationFiling(state) || isCorrectionFiling(state)) return OfficeTypes.REGISTERED_OFFICE
-}
-
 /** True if any office address has changed. Applies to corrections, change and conversion filings only. */
 export const hasOfficeAddressesChanged = (state: StateIF): boolean => {
   const isExpectedFiling = isCorrectionFiling(state) || isChangeFiling(state) || isConversionFiling(state)
-  const isAddressChanged = (mailingChanged(state) || deliveryChanged(state) ||
+  const hasMailingDeliveryChanged = hasMailingChanged(state) || hasDeliveryChanged(state)
+  const isChangeOrConversionFiling = isChangeFiling(state) || isConversionFiling(state)
+  const hasRecMailingDeliveryChanged = hasRecMailingChanged(state) || hasRecDeliveryChanged(state)
+  const hasAddressChanged = (
+    hasMailingDeliveryChanged ||
     // exclude Records Address conditions from Change or Conversion filing
-    (!isChangeFiling(state) && !isConversionFiling(state) && (recMailingChanged(state) || recDeliveryChanged(state)))
+    (!isChangeOrConversionFiling && hasRecMailingDeliveryChanged)
   )
 
-  return (isExpectedFiling && isAddressChanged)
+  return (isExpectedFiling && hasAddressChanged)
 }
 
 /** The office addresses from the original IA. NB: may be {} */
-export const originalOfficeAddresses = (state: StateIF): AddressesIF => {
+export const getOriginalOfficeAddresses = (state: StateIF): AddressesIF => {
   if (isCorrectionFiling(state)) {
     return (getOriginalIA(state)?.incorporationApplication.offices as AddressesIF)
   }
   if (isChangeFiling(state) || isConversionFiling(state)) {
-    return (getEntitySnapshot(state)?.addresses as AddressesIF)
+    return (getEntitySnapshot(state)?.addresses)
   }
 }
 
 /** True if (registered) mailing address has changed. */
-export const mailingChanged = (state: StateIF): boolean => {
-  return (
-    !isSame(getOfficeAddresses(state)?.[officeType(state)]?.mailingAddress,
-    originalOfficeAddresses(state)?.[officeType(state)]?.mailingAddress, ['addressCountryDescription'])
-  )
+export const hasMailingChanged = (state: StateIF): boolean => {
```

Comment:
As in other places, the different address scenarios are explicit here (instead of using "dynamic address type").